diff --git a/absl/copts/GENERATED_AbseilCopts.cmake b/absl/copts/GENERATED_AbseilCopts.cmake
index 51742c9..e59f8f9 100644
--- a/absl/copts/GENERATED_AbseilCopts.cmake
+++ b/absl/copts/GENERATED_AbseilCopts.cmake
@@ -131,6 +131,7 @@ list(APPEND ABSL_MSVC_FLAGS
     "/wd4267"
     "/wd4503"
     "/wd4800"
+    "/DNDEBUG"
 )
 
 list(APPEND ABSL_MSVC_LINKOPTS
diff --git a/absl/synchronization/mutex.cc b/absl/synchronization/mutex.cc
index 76ad41f..0b3bf4a 100644
--- a/absl/synchronization/mutex.cc
+++ b/absl/synchronization/mutex.cc
@@ -1410,12 +1410,7 @@ static GraphId DeadlockCheck(Mutex *mu) {
 // Invoke DeadlockCheck() iff we're in debug mode and
 // deadlock checking has been enabled.
 static inline GraphId DebugOnlyDeadlockCheck(Mutex *mu) {
-  if (kDebugMode && synch_deadlock_detection.load(std::memory_order_acquire) !=
-                        OnDeadlockCycle::kIgnore) {
-    return DeadlockCheck(mu);
-  } else {
     return InvalidGraphId();
-  }
 }
 
 void Mutex::ForgetDeadlockInfo() {
-- 
2.34.0.windows.1

